src/Raven.Server/Web/RequestHandler.cs

```diff
@@ -250,6 +250,27 @@ protected bool IsWebsocketRequest()
             return HttpContext.WebSockets.IsWebSocketRequest;
         }
 
+        protected string GetRaftRequestIdFromQuery()
+        {
+            var guid = GetStringQueryString("raft-request-id", required: false);
+
+            if (RequestRouter.TryGetClientVersion(HttpContext, out var clientVersion))
+            {
+                if (clientVersion.Build <= 42_008) // TODO: what should be the behaviour with the nightly version 
+                {
+                    guid = guid ?? string.Empty;
+                }
+            }
+
+#if DEBUG
+            return guid;
```

Comment:
I don't understand why you have this?

Reply:
If the version match, and no request id provided, should throw

Reply:
I want to avoid throwing any exception in `release`. 
In order not to break any client..

Reply:
Right now you aren't throwing anything anyway, though?

Reply:
If the guid is null, I do throw before appending to the log (both release and debug)
I want to throw only in debug builds, in release even if we have a bug we should put `string.Empty`.